neo/rawio/axonarawio.py

```diff
@@ -81,57 +98,149 @@ def _parse_header(self):
+                # 1st priority source: Spike2msMode (0 -> 48kHz; 1 -> 24kHz)
+                # 2nd priority source: tetrode sample rate
+                spikemode_to_sr = {0: 48000, 1: 24000}  # spikemode->rate in Hz
+                sm = self.file_parameters['set']['file_header'].get(
+                    'Spike2msMode', -1)
+                wf_sampling_rate = spikemode_to_sr.get(int(sm), None)
+                if wf_sampling_rate is None:
+                    wf_sampling_rate = self._to_hz(tdict['sample_rate'],
+                                               dtype=float)
+
+                spike_channels.append((unit_name, unit_id, wf_units, wf_gain,
+                                       wf_offset, wf_left_sweep,
+                                       wf_sampling_rate))
+
+                self.file_parameters['unit']['tetrode_ids'].append(i + 1)
+                self.file_parameters['unit'][i + 1] = tdict
+
+            # propagate common tetrode parameters to global unit level
+            units_dict = self.file_parameters['unit']
+            ids = units_dict['tetrode_ids']
+            copied_keys = []
+            if ids:
+                for key, value in units_dict[ids[0]].items():
+                    # copy key-value pair if present across all tetrodes
+                    if all([key in units_dict[t] for t in ids]) and \
+                            all([units_dict[t][key] == value for t in ids]):
+                        self.file_parameters['unit'][key] = value
+                        copied_keys.append(key)
+
+                # remove key from individual tetrode parameters
+                for key in copied_keys:
+                    for t in ids:
+                        self.file_parameters['unit'][t].pop(key)
+
+        # Create RawIO header dict
         self.header = {}
         self.header['nb_block'] = 1
         self.header['nb_segment'] = [1]
-        self.header['signal_streams'] = self._get_signal_streams_header()
-        self.header['signal_channels'] = self._get_signal_chan_header()
-        self.header['spike_channels'] = np.array([],
+        self.header['signal_streams'] = np.array(signal_streams,
+                                                 dtype=_signal_stream_dtype)
+        self.header['signal_channels'] = np.array(signal_channels,
+                                                  dtype=_signal_channel_dtype)
+        self.header['spike_channels'] = np.array(spike_channels,
                                                  dtype=_spike_channel_dtype)
         self.header['event_channels'] = np.array([],
                                                  dtype=_event_channel_dtype)
 
         # Annotations
         self._generate_minimal_annotations()
 
+        # Adding custom annotations
         bl_ann = self.raw_annotations['blocks'][0]
         seg_ann = bl_ann['segments'][0]
         seg_ann['rec_datetime'] = self.read_datetime()
-        sig_an = \
+        if seg_ann['signals']:
```

Comment:
maybe if len(seg_ann['signals']): is more easy to understand

Reply:
fixed